### PR TITLE
Add a guide for overriding assets

### PR DESCRIPTION
**Pull request by mamhoff**
This is pretty much a verbatim copy of the Spree guide on assets, which
I found to be comprehensive and well-done. It includes an Acknowledgements section
referencing the heroes who contributed to it.

### CODE REVIEW

**Comment by jhawthorn** [comment]
I'm actually not certain this works (anymore? after a certain sprockets version?).

When using require_tree, at least, JS assets aren't overridable (see #1567). @cbrunsdon would know more since he has run into this.

File: guides/assets.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+assets and how you can extend and customize them including:
+
+-   Understanding Solidus's use of the Rails asset pipeline
+-   Managing application specific assets
+-   Managing extension specific assets
+-   Overriding Solidus's core assets
+
+## Solidus's Asset Pipeline
+
+With the release of 3.1 Rails now supports powerful asset management
```
Comment (cbrunsdon) [inline guides/assets.md]: This makes Rails 3.1 sound new and fancy.

Can we change this to something :

> Solidus leverages [Rails' asset pipeline](http://guides.rubyonrails.org/asset_pipeline.html) to allow for extension and customization of frontend and backend assets.

File: guides/assets.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+features and Solidus fully leverages these features to further extend and
+simplify its customization potential. Using asset customization
+techniques outlined below you be able to adapt all the JavaScript,
+stylesheets and images contained in Solidus to easily provide a fully
+custom experience.
+
+All Solidus generated (or upgraded) applications include an `app/assets`
+directory (as is standard for all Rails 3.1 apps). We've taken this one
```
Comment (cbrunsdon) [inline guides/assets.md]: I'd remove the "3.1" here and just say Rails apps.

File: guides/assets.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+what the extension requires) manifest files, using the same directory
+structure as outlined above.
+
+Third party extension manifest files will not be automatically included
+in the relevant all.(js|css) files so it's important to document the
+manual inclusion in your extensions installation instructions or provide
+a Rails generator to do so.
+
+For an example of an extension using a generator to install assets and
+migrations take a look at the [install_generator for spree_fancy](https://github.com/spree/spree_fancy/blob/master/lib/generators/spree_fancy/install/install_generator.rb).
```
Comment (cbrunsdon) [inline guides/assets.md]: Why don't we use https://github.com/solidusio/solidus_auth_devise/blob/master/lib/generators/solidus/auth/install/install_generator.rb

File: guides/assets.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+what the extension requires) manifest files, using the same directory
+structure as outlined above.
+
+Third party extension manifest files will not be automatically included
+in the relevant all.(js|css) files so it's important to document the
+manual inclusion in your extensions installation instructions or provide
+a Rails generator to do so.
+
+For an example of an extension using a generator to install assets and
+migrations take a look at the [install_generator for solidus_auth_devise]( https://github.com/solidusio/solidus_auth_devise/blob/master/lib/generators/solidus/auth/install/install_generator.rb).
```
Comment (tvdeyen) [inline guides/assets.md]: the solidus_auth_devise generator actually don't install any assets ;)

So, maybe this one? https://github.com/solidusio/solidus_braintree/blob/master/lib/generators/solidus_braintree/install/install_generator.rb#L6-L13

**Pull request merged**